src/Workspaces/Core/Portable/Shared/TestHooks/IAsynchronousOperationListenerProvider.cs

```diff
@@ -178,8 +178,11 @@ public async Task WaitAllAsync(Workspace? workspace, string[]? featureNames = nu
                     eventProcessingAction?.Invoke();
 
                     // in unit test where it uses fake foreground task scheduler such as StaTaskScheduler
-                    // we need to yield for the scheduler to run inlined tasks
-                    await Task.Yield();
+                    // we need to yield for the scheduler to run inlined tasks. If we are not processing events, we
+                    // switch to the thread pool for the continuations since the yield will only let operations at the
+                    // same or higher priority to execute prior to the continuation.
+                    var continueOnCapturedContext = eventProcessingAction is object;
```

Comment:
As discovered in https://github.com/dotnet/roslyn/pull/48825, eventProcessingAction is always null, and so I'm deleting it in that PR. I'm happy to deal with the merge conflict if this goes in first.